client/src/controllers/BulkController.test.js

```diff
@@ -0,0 +1,58 @@
+import { Application } from '@hotwired/stimulus';
+import { BulkController } from './BulkController';
+
+describe('BulkController', () => {
+  beforeEach(() => {
+    document.body.innerHTML = `
+    <div data-controller="w-bulk">
+      <input type="checkbox" data-w-bulk-target="all" data-action="w-bulk#toggleAll">
+      <div>
+        <input type="checkbox" data-w-bulk-target="item" disabled data-action="w-bulk#toggle">
+        <input type="checkbox" data-w-bulk-target="item" data-action="w-bulk#toggle">
+        <input type="checkbox" data-w-bulk-target="item" data-action="w-bulk#toggle">
+      </div>
+      <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="false">Clear all</button>
+      <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="true">Select all</button>
+    </div>
+    `;
+    const application = Application.start();
+    application.register('w-bulk', BulkController);
+  });
+
+  it('executes the correct action when the select all checkbox is clicked', () => {
+    const allCheckbox = document.querySelector('[data-w-bulk-target="all"]');
+    const toggleAllActionSpy = jest.spyOn(
+      BulkController.prototype,
+      'toggleAll',
+    );
+
+    allCheckbox.click();
+
+    expect(toggleAllActionSpy).toHaveBeenCalled();
+  });
+
+  it('executes the correct action when an item checkbox is clicked', () => {
+    const itemCheckbox = document.querySelector(
+      '[data-w-bulk-target="item"]:not([disabled])',
+    );
+    const toggleActionSpy = jest.spyOn(BulkController.prototype, 'toggle');
+
+    itemCheckbox.click();
+
+    expect(toggleActionSpy).toHaveBeenCalled();
+  });
```

Comment:
```suggestion
  it('should keep the select all checkbox in sync when individual checkboxes are all ticked', () => {
    const allCheckbox = document.getElementById('select-all');
    expect(allCheckbox.checked).toBe(false);

    [
      ...document.querySelectorAll(
        '[data-w-bulk-target="item"]:not([disabled])',
      ),
    ].forEach((itemCheckbox) => {
      itemCheckbox.click();
    });

    expect(allCheckbox.checked).toBe(true);

    [
      ...document.querySelectorAll(
        '[data-w-bulk-target="item"]:not([disabled])',
      ),
    ].forEach((itemCheckbox) => {
      itemCheckbox.click();
    });

    expect(allCheckbox.checked).toBe(false);
  });
```

Similar here, we do not care about the methods that exist, we care that selecting each item checkbox individually should make the select all checkbox appear selected.